### PR TITLE
[link] polish api, link SEO & plugin data (RelValue & AnchorTarget)

### PR DESCRIPTION
**Pull request by maimonav**
- Refactor Link Components.
- Refactor link data usage (`relValue` & `anchorTarget`).
- Polish API:
Add `showNewTabCheckbox`, `showNoFollowCheckbox` & `showSponsoredCheckbox` to `LinkPanelSettings`.
Add `rel: Link_Rel` prop, that deprecates `relValue: string`.
- Link SEO: add Sponsored checkbox.
- Render rel `noopener noreferrer` as default value in all links.
- Deprecate `blankTargetToggleVisibilityFn`& `nofollowRelToggleVisibilityFn`.

### CODE REVIEW

File: packages/editor-common/web/src/Utils/draftUtils.ts [hunk 1]
```diff
@@ -677,7 +667,7 @@ export function fixPastedLinks(
       content.replaceEntityData(entityKey, {
         url,
         target: anchorTarget || '_self',
-        rel: relValue || 'noopener noreferrer',
+        rel: `noopener noreferrer ${rel}`,
```
Comment (maimonav) [inline packages/editor-common/web/src/Utils/draftUtils.ts:670]:
```suggestion
        rel,
```

File: packages/editor/web/src/RichContentEditor/Toolbars/LinkModal.jsx [hunk 2]
```diff
@@ -2,15 +2,15 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { LinkPanelContainer } from 'wix-rich-content-editor-common';
```
Comment (maimonav) [inline packages/editor/web/src/RichContentEditor/Toolbars/LinkModal.jsx:4]: This class can be deleted

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx [hunk 3]
```diff
@@ -202,13 +202,14 @@ const createBaseComponent = ({
...
       }
-      const { url, anchor, target, rel } = linkData;
+      const { url, anchor, target, rel, sponsored } = linkData;
       if (this.isMeAndIdle()) {
         const link = url
           ? {
               url,
               target,
               rel,
+              sponsored,
```
Comment (maimonav) [inline packages/plugin-commons/web/src/Base/createBaseComponent.jsx:212]: Delete this changes

File: packages/plugin-commons/web/src/Base/toolbars/createAtomicPluginToolbar.jsx [hunk 4]
```diff
@@ -111,12 +111,13 @@ export default function createAtomicPluginToolbar({
...
         return;
       }
-      const { url, anchor, target, rel } = linkData;
+      const { url, anchor, target, rel, sponsored } = linkData;
       const link = url
         ? {
             url,
             target,
             rel,
+            sponsored,
```
Comment (maimonav) [inline packages/plugin-commons/web/src/Base/toolbars/createAtomicPluginToolbar.jsx:120]: Delete this changes

File: packages/plugin-gallery/web/src/components/gallery-controls/gallery-image-settings.js [hunk 5]
```diff
@@ -56,7 +56,7 @@ class ImageSettings extends Component {
     this.props.onUpdateItem({ link: this.linkPanelToLink(linkPanelValues) });
   };
 
-  linkPanelToLink = ({ url, targetBlank, nofollow, isValid }) => ({
+  linkPanelToLink = ({ url, targetBlank, nofollow, isValid, sponsored }) => ({
```
Comment (maimonav) [inline packages/plugin-gallery/web/src/components/gallery-controls/gallery-image-settings.js:59]: fix this

File: packages/ricos-common/web/src/RicosEngine.tsx [hunk 6]
```diff
@@ -7,6 +7,7 @@ import { merge } from 'lodash';
 import previewStrategy from './previewStrategy/previewStrategy';
 import { PreviewConfig } from 'wix-rich-content-preview';
 import { RicosEditorProps, RicosViewerProps, RichContentProps, BasePlugin } from './types';
+import { convertRelStringToObject } from 'wix-rich-content-common';
```
Comment (maimonav) [inline packages/ricos-common/web/src/RicosEngine.tsx:11]: Fix bundle size

File: packages/editor-common/web/src/Components/LinkComponents/BasicLinkPanel.jsx [hunk 7]
```diff
@@ -51,13 +52,14 @@ class BasicLinkPanel extends PureComponent {
...
       >
-        {isMobile && linkPanelWithTitle && this.renderMobileTitle()}
+        {isMobile && this.renderMobileTitle()}
         <div className={styles.linkPanel_content}>
           <LinkPanel
             linkValues={linkPanelValues}
-            onChange={linkPanelValues => onChangeLinkPanel({ linkPanelValues })}
-            showTargetBlankCheckbox={showTargetBlankCheckbox}
-            showRelValueCheckbox={showRelValueCheckbox}
+            onChange={linkPanelValues => onChangeLinkPanel(linkPanelValues)}
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/LinkComponents/BasicLinkPanel.jsx:59]:
```suggestion
            onChange={onChangeLinkPanel}
```

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx [hunk 8]
```diff
@@ -41,11 +43,15 @@ class LinkPanel extends Component {
...
+    this.onChange({ rel: { nofollow: event.target.checked } });
+  };
+
+  handleSponsoredChange = event => {
+    this.onChange({ rel: { sponsored: event.target.checked } });
   };
 
   onChange = changes => {
-    this.props.onChange({ ...this.props.linkValues, ...changes });
+    this.props.onChange({ ...merge(this.props.linkValues, changes) });
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx:54]:
```suggestion
    this.props.onChange(merge(this.props.linkValues, changes));
```

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx [hunk 9]
```diff
@@ -112,15 +118,17 @@ class LinkPanel extends Component {
...
+      showSponsoredCheckbox,
       t,
       linkValues,
       unchangedUrl,
       isMobile,
     } = this.props;
 
-    const { targetBlank, nofollow } = linkValues;
+    const { targetBlank, rel } = linkValues;
+    const { nofollow, sponsored /*, ugc*/ } = rel || {};
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx:131]:
```suggestion
    const { nofollow, sponsored } = rel || {};
```

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx [hunk 10]
```diff
@@ -157,6 +165,17 @@ class LinkPanel extends Component {
...
           )}
+          {showSponsoredCheckbox && (
+            <Checkbox
+              label={t('LinkPanel_Sponsored_Checkbox')}
+              theme={theme}
+              checked={sponsored}
+              dataHook="linkPanelSponsoredCheckbox"
+              onChange={this.handleSponsoredChange}
+              t={t}
+              isMobile={isMobile}
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx:176]: NewTabCheckbox doesn't have isMobile. Probably needed in all or none of the checkboxes.

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx [hunk 11]
```diff
@@ -157,6 +165,17 @@ class LinkPanel extends Component {
...
             />
           )}
+          {showSponsoredCheckbox && (
+            <Checkbox
+              label={t('LinkPanel_Sponsored_Checkbox')}
+              theme={theme}
+              checked={sponsored}
+              dataHook="linkPanelSponsoredCheckbox"
+              onChange={this.handleSponsoredChange}
+              t={t}
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx:175]: is needed?

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanelContainer.jsx [hunk 12]
```diff
@@ -9,27 +9,17 @@ import { isEmpty } from 'lodash';
...
-      originalLinkPanel,
-    } = this.props;
+    const { url, anchor, targetBlank, rel, editorState, linkTypes, unchangedUrl } = this.props;
     this.renderBasicLinkPanel =
       !linkTypes ||
       isEmpty(linkTypes) ||
       !Object.values(linkTypes).find(addon => !!addon) ||
-      unchangedUrl ||
-      originalLinkPanel;
+      unchangedUrl;
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/LinkComponents/LinkPanelContainer.jsx:17]:
```suggestion
      hideUrlInput;
```

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/editor-common/web/src/Components/LinkComponents/BasicLinkPanel.jsx [hunk 13]
```diff
@@ -51,13 +52,14 @@ class BasicLinkPanel extends PureComponent {
         role="form"
         {...ariaProps}
       >
-        {isMobile && linkPanelWithTitle && this.renderMobileTitle()}
+        {isMobile && this.renderMobileTitle()}
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/editor-common/web/src/Components/LinkComponents/BasicLinkPanel.jsx:54]: Always true so I removed it

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx (hunk 10 above)
Comment (maimonav) [reply packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx:176]: `isMobile` needed only for checkboxes with tooltips.

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx (hunk 11 above)
Comment (maimonav) [reply packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx:175]: Nope

File: packages/editor-common/web/src/Components/LinkComponents/BasicLinkPanel.jsx (hunk 7 above)
Comment (maimonav) [reply packages/editor-common/web/src/Components/LinkComponents/BasicLinkPanel.jsx:58]: Done.

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanelContainer.jsx (hunk 12 above)
Comment (maimonav) [reply packages/editor-common/web/src/Components/LinkComponents/LinkPanelContainer.jsx:17]: Done.

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx (hunk 9 above)
Comment (maimonav) [reply packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx:131]: Done.

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx (hunk 8 above)
Comment (maimonav) [reply packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx:54]: `merge` of lodash makes in-place changes in the destination object, `this.props.linkValues` here, which lead to a weird behavior, so I fixed it by making a new object to be the destination object.

File: examples/main/shared/editor/EditorPlugins.tsx [hunk 14]
```diff
@@ -355,6 +355,7 @@ export const uiSettings: UISettings = {
     externalPopups: true,
     blankTargetToggleVisibilityFn: () => true,
     nofollowRelToggleVisibilityFn: () => true,
+    showSponsoredCheckbox: true,
```
[the remark this answers is not in the document]
Comment (maimonav) [reply examples/main/shared/editor/EditorPlugins.tsx:358]: Done.

File: e2e/tests/__snapshots__/media.e2e.js.snap [hunk 15]
```diff
@@ -3405,3 +3407,314 @@ exports[`plugins > youTube > open youTube modal #0`] =
   ],
   "entityMap": {}
 };
+
+exports[`plugins > image > should disable image expand #0`] =
```
Comment (shaulgo) [inline e2e/tests/__snapshots__/media.e2e.js.snap:3411]: please revert

File: packages/editor/web/src/RichContentEditor/Toolbars/BlockLinkModal.jsx [hunk 16]
```diff
@@ -56,36 +58,33 @@ export default class BlockLinkModal extends Component {
...
-      relValue,
       t,
       uiSettings,
-      unchangedUrl,
+      hideUrlInput,
       linkTypes,
       editorState,
     } = this.props;
     const componentLink = pubsub.get('componentData')?.config?.link;
     const { url, anchor, target, rel } = componentLink || {};
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/Toolbars/BlockLinkModal.jsx:68]:
```suggestion
    const { url, anchor, target = anchorTaget, rel } = componentLink || {};
```

File: packages/common/web/src/Utils/linkConverters.ts [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  return relObject;
+};
+
+export const convertTargetStringToBoolean = (target?: string, anchorTarget?: string) =>
+  target === BLANK || anchorTarget === BLANK;
+
+export const convertTargetBooleanToString = (targetBlank?: boolean, anchorTarget?: string) => {
+  if (targetBlank) {
```
Comment (shaulgo) [inline packages/common/web/src/Utils/linkConverters.ts:54]: check if anchorTarget should be default value of target

File: packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelDesktop.jsx [hunk 18]
```diff
@@ -74,9 +75,10 @@ class MultiSelectLinkPanelDesktop extends PureComponent {
             <div className={styles.multiSelectLinkPanel_LinkPanelContainer}>
               <LinkPanel
                 linkValues={linkPanelValues}
-                onChange={linkPanelValues => onChangeLinkPanel({ linkPanelValues })}
-                showTargetBlankCheckbox={showTargetBlankCheckbox}
-                showRelValueCheckbox={showRelValueCheckbox}
+                onChange={linkPanelValues => onChangeLinkPanel(linkPanelValues)}
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelDesktop.jsx:78]:
```suggestion
                onChange={onChangeLinkPanel}
```

File: packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelDesktop.jsx [hunk 19]
```diff
@@ -85,7 +87,7 @@ class MultiSelectLinkPanelDesktop extends PureComponent {
             <AnchorPanel
               anchorableBlocksData={anchorableBlocksData}
               anchorValues={anchorPanelValues}
-              onChange={anchorPanelValues => onChangeAnchorPanel({ anchorPanelValues })}
+              onChange={anchorPanelValues => onChangeAnchorPanel(anchorPanelValues)}
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelDesktop.jsx:90]:
```suggestion
              onChange={onChangeAnchorPanel}
```

File: packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelMobile.jsx [hunk 20]
```diff
@@ -72,7 +74,7 @@ class MultiSelectLinkPanelMobile extends PureComponent {
             <AnchorPanel
               anchorableBlocksData={anchorableBlocksData}
               anchorValues={anchorPanelValues}
-              onChange={anchorPanelValues => onChangeAnchorPanel({ anchorPanelValues })}
+              onChange={anchorPanelValues => onChangeAnchorPanel(anchorPanelValues)}
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelMobile.jsx:77]:
```suggestion
              onChange={onChangeAnchorPanel}
```

File: packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelMobile.jsx [hunk 21]
```diff
@@ -61,9 +62,10 @@ class MultiSelectLinkPanelMobile extends PureComponent {
             <div className={styles.multiSelectLinkPanel_LinkPanelContainer_mobile}>
               <LinkPanel
                 linkValues={linkPanelValues}
-                onChange={linkPanelValues => onChangeLinkPanel({ linkPanelValues })}
-                showTargetBlankCheckbox={showTargetBlankCheckbox}
-                showRelValueCheckbox={showRelValueCheckbox}
+                onChange={linkPanelValues => onChangeLinkPanel(linkPanelValues)}
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelMobile.jsx:65]:
```suggestion
                onChange={onChangeLinkPanel}
```

File: packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelDesktop.jsx [hunk 22]
```diff
@@ -74,9 +75,10 @@ class MultiSelectLinkPanelDesktop extends PureComponent {
             <div className={styles.multiSelectLinkPanel_LinkPanelContainer}>
               <LinkPanel
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelDesktop.jsx:76]: if super easy to merge mobile and desktop components then do it

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/common/web/src/Utils/linkConverters.ts (hunk 17 above)
Comment (maimonav) [reply packages/common/web/src/Utils/linkConverters.ts:54]: Yes

File: packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelDesktop.jsx (hunk 18 above)
Comment (maimonav) [reply packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelDesktop.jsx:78]: Done.

File: packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelDesktop.jsx (hunk 19 above)
Comment (maimonav) [reply packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelDesktop.jsx:90]: Done.

File: packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelMobile.jsx (hunk 21 above)
Comment (maimonav) [reply packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelMobile.jsx:65]: Done.

File: packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelDesktop.jsx (hunk 22 above)
Comment (maimonav) [reply packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanelDesktop.jsx:76]: Done.

File: packages/editor/web/src/RichContentEditor/Toolbars/BlockLinkModal.jsx (hunk 16 above)
Comment (maimonav) [reply packages/editor/web/src/RichContentEditor/Toolbars/BlockLinkModal.jsx:68]: Done.

File: e2e/tests/__snapshots__/media.e2e.js.snap (hunk 15 above)
Comment (maimonav) [reply e2e/tests/__snapshots__/media.e2e.js.snap]: Done.

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/common/web/src/Utils/linkConverters.ts [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+};
+
+export const convertTargetStringToBoolean = (target?: string) => target === BLANK;
+
+export const convertTargetBooleanToString = (targetBlank?: boolean, anchorTarget?: string) => {
+  if (targetBlank) {
+    return BLANK;
+  } else if (targetBlank === false) {
+    return SELF;
+  } else return anchorTarget || SELF;
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/common/web/src/Utils/linkConverters.ts]: Done.

File: packages/plugin-button/web/src/components/settings-component.jsx [hunk 24]
```diff
@@ -38,24 +45,33 @@ class SettingsComponent extends PureComponent {
...
 
-  handleRelChange = event => {
-    this.setState({ rel: event.target.checked });
+  onLinkPanelChange = linkPanelValues => {
+    this.setState({ ...this.linkPanelToLink(linkPanelValues) });
   };
 
   render() {
-    const { t, linkInputRef, isMobile, validUrl, shouldShowLink } = this.props;
+    const { t, shouldShowLink, uiSettings, theme } = this.props;
```
Comment (shaulgo) [inline packages/plugin-button/web/src/components/settings-component.jsx:70]:
```suggestion
    const { t, shouldShowLinkPanel, uiSettings, theme } = this.props;
```

File: packages/plugin-link/web/src/toolbar/UrlLinkButton.jsx [hunk 25]
```diff
@@ -37,7 +37,7 @@ export default class UrlLinkButton extends Component {
     const anchorProps = {
       href,
       target: target || '_self',
-      rel: rel || 'noopener',
+      rel: `noopener noreferrer ${rel || ''}`.trim(),
```
Comment (shaulgo) [inline packages/plugin-link/web/src/toolbar/UrlLinkButton.jsx:40]: fix

File: packages/ricos-content/web/src/normalization/dataNormalizers/linkDataNormalizer.ts [hunk 26]
```diff
@@ -16,7 +16,7 @@ const linkDataNormalizer = (
 
   return {
     target: targetBlank ? '_blank' : anchorTarget || '_self',
```
Comment (shaulgo) [inline packages/ricos-content/web/src/normalization/dataNormalizers/linkDataNormalizer.ts:18]:
```suggestion
    target: targetBlank ? '_blank' : anchorTarget || '_top',
```

File: website/docs/ricos/ricos-api.mdx [hunk 27]
```diff
@@ -194,22 +194,44 @@ Similar to [content](#content) prop; allows content to be injected to mounted Ri
 
 ```jsx
 linkPanelSettings?: {
-  blankTargetToggleVisibilityFn?: (anchorTarget: string) => boolean;
-  nofollowRelToggleVisibilityFn?: (relValue: string) => boolean;
+  showNewTabCheckbox?: boolean;
+  showNoFollowCheckbox?: boolean;
+  showSponsoredCheckbox?: boolean;
+  blankTargetToggleVisibilityFn?: () => boolean;
+  nofollowRelToggleVisibilityFn?: () => boolean;
```
Comment (shaulgo) [inline website/docs/ricos/ricos-api.mdx:201]: delete

File: website/docs/ricos/ricos-api.mdx [hunk 28]
```diff
@@ -194,22 +194,44 @@ Similar to [content](#content) prop; allows content to be injected to mounted Ri
...
+By default, it's visible.
+
+#### `showSponsoredCheckbox`
+
+Determines the 'Mark as a sponsored link' link panel checkbox visiblity.
+By default, it's visible.
+
 #### `blankTargetToggleVisibilityFn`
 
+Deprecated, please use `showNewTabCheckbox` instead.
```
Comment (shaulgo) [inline website/docs/ricos/ricos-api.mdx:224]:
```suggestion
*** Deprecated *** please use `showNewTabCheckbox` instead.
```

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/plugin-button/web/src/components/settings-component.jsx (hunk 24 above)
Comment (maimonav) [reply packages/plugin-button/web/src/components/settings-component.jsx:70]: Done

File: packages/plugin-link/web/src/toolbar/UrlLinkButton.jsx (hunk 25 above)
Comment (maimonav) [reply packages/plugin-link/web/src/toolbar/UrlLinkButton.jsx:40]: Done.

File: website/docs/ricos/ricos-api.mdx (hunk 27 above)
Comment (maimonav) [reply website/docs/ricos/ricos-api.mdx:201]: Done

File: packages/common/web/src/Utils/linkConverters.ts [hunk 29]
```diff
@@ -0,0 +1,64 @@
...
+  if (rel.includes('ugc')) {
+    relObject.ugc = true;
+  }
+
+  return relObject;
+};
+
+export const convertTargetStringToBoolean = (target?: string) => target === BLANK;
+
+export const convertTargetBooleanToString = (targetBlank?: boolean, anchorTarget?: string) => {
```
Comment (shaulgo) [inline packages/common/web/src/Utils/linkConverters.ts:52]: unit test

File: packages/plugin-button/web/src/components/settings-component.jsx [hunk 30]
```diff
@@ -38,24 +45,33 @@ class SettingsComponent extends PureComponent {
...
+      rel: convertRelObjectToString(rel),
+    };
+  };
+
+  linkToLinkPanel = ({ url, target, rel }) => {
+    return {
+      url,
+      targetBlank: convertTargetStringToBoolean(target),
+      rel: convertRelStringToObject(rel),
+    };
```
Comment (shaulgo) [inline packages/plugin-button/web/src/components/settings-component.jsx:62]: Move to converters file

File: packages/plugin-link/web/src/LinkComponent.jsx [hunk 31]
```diff
@@ -2,27 +2,13 @@ import React from 'react';
...
-  href,
-  rel,
-  target,
-  ...otherProps
-}) => {
-  const componentData = href
-    ? { url: href, rel, target }
-    : contentState.getEntity(entityKey).getData();
+const Link = ({ entityKey, contentState, children, anchorTarget, relValue, ...otherProps }) => {
+  const componentData = contentState.getEntity(entityKey).getData();
```
Comment (shaulgo) [inline packages/plugin-link/web/src/LinkComponent.jsx:6]: 1. check if we can use this line in createLinkPlugin. 
2. delete this file

File: packages/plugin-link/web/src/LinkViewer.jsx [hunk 32]
```diff
@@ -67,21 +69,13 @@ class LinkViewer extends Component {
...
   render() {
-    const { componentData, anchorTarget, relValue, children, isInEditor } = this.props;
-    const { url, anchor, target, rel } = componentData;
+    const { componentData, anchorTarget, children, isInEditor } = this.props;
+    const { url, anchor, target = anchorTarget, rel } = componentData;
     const anchorProps = {
       href: this.getHref(url, anchor),
-      target: this.getTarget(anchor, target, anchorTarget),
-      rel: rel ? rel : relValue || 'noopener',
+      target: anchor ? '_top' : getTargetValue(target),
```
Comment (shaulgo) [inline packages/plugin-link/web/src/LinkViewer.jsx:77]:
```suggestion
      target: anchor ? '_self' : getTargetValue(target),
```

File: packages/ricos-editor/web/src/utils/editorCommandsTestsUtil.ts [hunk 33]
```diff
@@ -487,18 +487,18 @@ const image = {
 const link = {
   type: RICOS_LINK_TYPE,
   decorationType: Decoration_Type.LINK,
-  data1: LinkData.fromJSON({ url: 'www.wix.com' }),
+  data1: LinkData.fromJSON({ url: 'www.wix.com', rel: 'nofollow', target: '_self' }),
```
Comment (shaulgo) [inline packages/ricos-editor/web/src/utils/editorCommandsTestsUtil.ts:490]: need to check

File: packages/viewer/web/src/getPluginViewers.jsx [hunk 34]
```diff
@@ -112,19 +114,19 @@ class PluginViewer extends PureComponent {
...
+            rel: getRelValue(rel),
           };
         }
         if (hasAnchor) {
           const { anchor } = config.link;
           const href = `#viewer-${anchor}`;
           containerProps = {
             href,
-            target: '_self',
+            target: '_top',
```
Comment (shaulgo) [inline packages/viewer/web/src/getPluginViewers.jsx:129]:
```suggestion
            target: '_self',
```

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/plugin-link/web/src/LinkComponent.jsx (hunk 31 above)
Comment (maimonav) [reply packages/plugin-link/web/src/LinkComponent.jsx:6]: Done.

File: packages/plugin-link/web/src/LinkViewer.jsx (hunk 32 above)
Comment (maimonav) [reply packages/plugin-link/web/src/LinkViewer.jsx:77]: Done.

File: packages/viewer/web/src/getPluginViewers.jsx (hunk 34 above)
Comment (maimonav) [reply packages/viewer/web/src/getPluginViewers.jsx:129]: Done.

File: packages/common/web/src/test/linkConverters.spec.ts [hunk 35]
```diff
@@ -0,0 +1,80 @@
...
+const relObject = { nofollow: true, sponsored: true, ugc: true };
+const target1 = '_blank';
+const target2 = '_top';
+const targetBlank1 = true;
+const targetBlank2 = false;
+const anchorTarget = '_blank';
+const defaultRelValue = 'noopener noreferrer';
+const dynamicRelValue = `${defaultRelValue} ${relString}`;
+
+const compare = (relObject1, relObject2) =>
```
Comment (shaulgo) [inline packages/common/web/src/test/linkConverters.spec.ts:20]: can probably be replaced with toEqual

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/common/web/src/test/linkConverters.spec.ts (hunk 35 above)
Comment (maimonav) [reply packages/common/web/src/test/linkConverters.spec.ts:20]: Done.

**Review by shaulgo — APPROVED** [review]

**Pull request merged**